tools/tsp-client/src/index.ts

```diff
@@ -1,62 +1,217 @@
-  const tspUrl = await tryReadTspLocation(outputDir);
-  if (!tspUrl) {
-    throw new Error("No tsp-location.yaml found, and no main file specified");
+// async function discoverMainFileUrl(outputDir: string): Promise<string> {
+//   const tspUrl = await tryReadTspLocation(outputDir);
+//   if (!tspUrl) {
+//     throw new Error("No tsp-location.yaml found, and no main file specified");
+//   }
+//   Logger.debug(`Using base url from  tsp-location.yaml: ${tspUrl}`);
+//   for (const mainFile of ["client.tsp", "main.tsp"]) {
+//     let url = new URL(mainFile, tspUrl).toString();
+//     Logger.debug(`Checking for file: ${url}`);
+//     if (await doesFileExist(url)) {
+//       Logger.debug(`Found main file: ${url}`);
+//       return url;
+//     }
+//   }
+//   throw new Error(`No main.tsp or client.tsp found for base url ${tspUrl}`);
+// }
+
+async function getEmitterOptions(rootUrl: string, emitter: string): Promise<string> {
+  // TODO: Add a way to specify emitter options like Language-Settings.ps1, could be a languageSettings.ts file
+  // Method signature should just include the rootUrl. Everything else should be included in the language-settings.yaml file
+  return `--options ${emitter}.emitter-output-dir=${rootUrl}`;
+}
+
+async function discoverMainFile(srcDir: string): Promise<string> {
+  Logger.debug(`Discovering entry file in ${srcDir}`)
+  var entryTsp = "";
+  const files = await readdir(srcDir, {recursive: true });
+  for (const file of files) {
+    if (file.includes("client.tsp") || file.includes("main.tsp")) {
+      entryTsp = file;
+      Logger.debug(`Found entry file: ${entryTsp}`);
+      return entryTsp;
+    }
+  };
+  throw new Error(`No main.tsp or client.tsp found`);
+}
+
+// async function downloadTspFiles(rootUrl: string, emitter: string, outputDir: string) {
+//   const { moduleImports, fileTree } = await downloadTsp(rootUrl);
+//   const { mainFilePath, files } = await fileTree.createTree();
+//   const tempRoot = await createTempDirectory(outputDir);
+//   const srcDir = path.join(tempRoot, "src");
+//   await writeFileTree(srcDir, files);
+
+//   const emitterPackage = getEmitterPackage(emitter);
+//   moduleImports.add(emitterPackage);
+//   await createPackageJson(tempRoot, moduleImports);
+//   return {
+//     tempRoot: tempRoot,
+//     srcDir: srcDir,
+//     mainFilePath: mainFilePath,
+//     emitterPackage: emitterPackage
+//   };
+// }
+
+async function sdkInit(
+  {
+    config,
+    outputDir,
+    // commit, //TODO
+    // repoUrl, // TODO
+  }: {
+    config: string;
+    outputDir: string;
+    // commit: string;
+    // repoUrl: string;
+  }): Promise<string> {
+  if (await doesFileExist(config)) {
+    // URL scenario
+    const tspConfigUrl = new URL(config).toString();
+    const matchRes = tspConfigUrl.match('^https://(?<urlRoot>github).com/(?<repo>[^/]*/azure-rest-api-specs(-pr)?)/blob/(?<commit>[0-9a-f]{40})/(?<path>.*)/tspconfig.yaml$')
```

Comment:
yuck, can we abstract this into a helper that parses the URL if we're going to go down this path?